tests/core/test_test.py

```diff
@@ -870,3 +870,41 @@ def test_test_generation_with_timestamp(tmp_path: Path) -> None:
     assert test["test_foo"]["outputs"] == {
         "query": [{"ts_col": datetime.datetime(2024, 9, 20, 11, 30, 0, 123456)}]
     }
+
+
+def test_test_generation_with_nested_arrays(tmp_path: Path) -> None:
```

Comment:
Let's just update `test_test_generation_with_array` instead of defining a new one here?